addons/website/models/website.py

```diff
@@ -506,6 +501,42 @@ def is_user(self):
     def is_public_user(self):
         return request.env.user.id == request.website.user_id.id
 
+    @api.model
+    def viewref(self, view_id, raise_if_not_found=True):
+        ''' Given an xml_id or a view_id, return the corresponding view record.
+            In case of website context, return the most specific one.
+            :param view_id: either a string xml_id or an integer view_id
+            :param raise_if_not_found: should the method raise an error if no view found
+            :return: The view record or empty recordset
+        '''
+        View = self.env['ir.ui.view']
+        view = None
+        if isinstance(view_id, pycompat.string_types):
+            if 'website_id' in self._context:
+                domain = [('key', '=', view_id)] + self.env['website'].website_domain(self._context.get('website_id'))
+                order = 'website_id'
+            else:
+                domain = [('key', '=', view_id)]
+                order = View._order
+            views = View.with_context(active_test=False).search(domain, order=order)
+            if views:
+                view = views.filter_duplicate()
+            else:
+                view = self.env.ref(view_id)
+                # self.env.ref might return something else than an ir.ui.view (eg: a theme.ir.ui.view)
+                if view._name != 'ir.ui.view':
+                    view = None
+        elif isinstance(view_id, pycompat.integer_types):
+            view = View.browse(view_id)
+        else:
+            raise ValueError('Expecting a string or an integer, not a %s.' % (type(view_id)))
```

Comment:
trad?

Reply:
Majority of ValueError are not translated in Odoo's code, should I make it translatable?
Note that the exact same error from self.env.ref's xmlid_to_object is also not translatable: https://github.com/odoo-dev/odoo/blame/768d3a6d7b0e1cd7e62c97e9b2a69420861b9873/odoo/addons/base/models/ir_model.py#L1396